test/Microsoft.OpenApi.Readers.Tests/V31Tests/ReferenceSamples/localReferenceToJsonSchemaResource.yaml

```diff
@@ -0,0 +1,23 @@
+openapi: 3.1.0
+info:
+  title: OpenAPI document containing examples for reuse
+  version: 1.0.0
+components:
+  schemas:
+    a:
+      type:
+        - object
+        - 'null'
+      properties:
+        b:
+          type:
+            - object
+            - 'null'
+          properties:
+            c:
+              type:
+                - object
+                - 'null'
+              properties:
+                b:
+                 $ref: '#/properties/b'
```

Comment:
This one looks invalid to me.
If I go back up one "schema level", I'm within the b schema definition, (L12), and there's no b property there. Am I reading it wrong?

Reply:
I'm getting it the same way, but this is the same example given in the [spec](https://github.com/MicrosoftDocs/openapi-docs/blob/c2b8bfcee7340f62efbe9bb87b9b68c3d0278afb/OpenAPI/OpenAPI.NET/references-openapi.md?plain=1#L270-L290) and [Safia's example](https://github.com/microsoft/OpenAPI.NET/issues/2062). 
So in your opinion the JSON pointer should be `#/a/properties/b` since a has a b property defined within it?

Maybe @darrelmiller can offer more insight?

Reply:
@MaggieKimani1 @baywet You don't "go back up one schema level".  You go back up until you reach a "JSON Schema Resource".  This is either a schema with a $Id, the root of the document, or in the case of OpenAPI to the component schema.  In this case "a" is the JSON Schema Resource which is effectively the root document as far as JSON Schema is concerned. The label "a" is outside of the JSON Schema Resource which is why it is not part of the $ref.

Reply:
Thanks for clarifying. I made a similar comment on the spec side PR. I think that "routine" should be explicitly called out there as it's not "broadly known"